main.go

```diff
@@ -1,235 +1,237 @@
-	fmt.Fprintf(conn, "window.location='%s'\n", program.URL)
-	status, err := bufio.NewReader(conn).ReadString('\n')
-	fmt.Printf("  %s", status)
-
-	select {
-	case <-time.After(program.Duration):
-		// Do nothing.
-		pause = false
-	case <-abort:
-		fmt.Println("Current program aborted")
-		return
-	}
-}
-
-func LoadAndRunLoop() {
-
-	// Load and run the acctive program_file indefinately
-	for {
-		// We pull filename inside the loop because the
-		// configuration can change while our program is running.
-		filename := viper.GetString("program_file")
-		for pause == true {
-			fmt.Println("Paused.")
-			time.Sleep(1 * time.Second)
-		}
-
-		pl := loadProgramList(filename)
-
-		for _, p := range pl {
-			for pause == true {
-				fmt.Println("Program list is paused.")
-				time.Sleep(1 * time.Second)
-			}
-			runProgram(p)
-		}
-
-		fmt.Printf("\nLooping back to play program list from beginning\n\n")
-	}
-
-}
-
-func PlayHandler(w http.ResponseWriter, r *http.Request) {
-
-	r.ParseForm()
-	var p Program
-	p.URL = r.Form.Get("url")
-	fmt.Printf("URL: %s\n", p.URL)
-
-	d := r.Form.Get("duration")
-	fmt.Printf("Duration: %s\n", d)
-
-	var err error
-	p.Duration, err = time.ParseDuration(r.Form.Get("duration"))
-	if err != nil {
-		w.Write([]byte("Program rejected.  Invalid duration.\n"))
-		return
-	}
-
-	// Needs validation...
-
-	// Now do something with the program.. play it?
-
-	// Stop normal rotation
-	pause = true
-	abort <- struct{}{}
-
-	go runProgram(p)
-	w.Write([]byte("Program accepted\n"))
-
-}
-
-func PauseHandler(w http.ResponseWriter, r *http.Request) {
-	pause = true
-	log.Println("Pausing from web request")
-	w.Write([]byte("Ok, paused.\n"))
-}
-
-func ResumeHandler(w http.ResponseWriter, r *http.Request) {
-	pause = false
-	log.Println("Unpausing from web request")
-	w.Write([]byte("Ok, unpaused.\n"))
-}
-func SkipHandler(w http.ResponseWriter, r *http.Request) {
-	log.Println("Skippingfrom web request")
-	pause = false
-	abort <- struct{}{}
-
-	w.Write([]byte("Skipping current programming and resume program list runner from web request.\n"))
-}
-
-// Control channel to stop running programs immediately (yes, global)
-var abort = make(chan struct{})
-var pause bool = false
-
-func main() {
-
-	InitializeConfig()
-
-	go LoadAndRunLoop()
-
-	r := mux.NewRouter()
-	r.HandleFunc("/play", PlayHandler)
-	r.HandleFunc("/pause", PauseHandler)
-	r.HandleFunc("/resume", ResumeHandler)
-	r.HandleFunc("/skip", SkipHandler)
-
-	go log.Fatal(http.ListenAndServe(":8080", r))
+// package main
+
+// import (
+// 	"bufio"
+// 	"encoding/csv"
+// 	"fmt"
+// 	"io"
+// 	"io/ioutil"
+// 	"log"
+// 	"net"
+// 	"net/http"
+// 	"os"
+// 	"strings"
+// 	"time"
+
+// 	"github.com/fsnotify/fsnotify"
+// 	"github.com/gorilla/mux"
+// 	"github.com/spf13/viper"
+// )
+
+// type Program struct {
+// 	URL      string
+// 	Duration time.Duration
+// }
+
+// // InitializeConfig loads our configuration using Viper package.
+// func InitializeConfig() {
```

Comment:
I think you are telling me that you like the name LoadConfig better than the name InitializeConfig.   Is that right?

FWIW,  I only picked the latter because it seemed to be the convention of the Viper author (as seen in the Hugo source code: https://github.com/spf13/hugo/blob/master/commands/hugo.go#L279

If you were telling me something else, I'm sorry, I missed your point.